whylog/config/__init__.py

```diff
@@ -13,6 +16,111 @@
 from whylog.config.rule import RegexRuleFactory
 
 
+class AbstractConfigFactory(object):
+    """
+    This factory is responsible for finding .whylog directory (whylog config directory) and basing on
+    found content directory creating object of subclass AbstractConfig. If not found then it creates minimal
+    .whylog version in current directory.
+    """
+    WHYLOG_DIR = '.whylog'
+    HOME_DIR = os.path.expanduser('~')
+    ETC_DIR = '/etc'
+    ASSISTANTS_DICT = {'regex': RegexAssistant}
+
+    @classmethod
+    @abstractmethod
+    def load_config(cls, path):
+        raise NotImplementedError
```

Comment:
`pass` instead of raising, so that super() calls in subclasses don't have a problem